CONTRIBUTING.md

```diff
@@ -0,0 +1,53 @@
+# Contributing
+
+## Are you writing a migration to change Publishing API data?
+
+We have a lot of migrations added to the Publishing API to change data, our
+long term plan is to reduce these so that these can be done via the API itself
+and not need migrations. Until we are at that point please use the following
+guidelines when creating migrations.
+
+### Run the migration on your machine first
+
+At the very least this will ensure that the code runs and that your data looks
+correct, but it also allows you to get an idea for how long the migration will
+take to finish. You should then include that information in the PR to help with
+whoever is doing the deployment.
+
+We also encourage you to run and check the results of the migration on
+integration, since the data there will be closer to production.
+
+### Always include the `schema.rb` file
+
+Even if the change is just the timestamp, it's important to include this file
+in your commit otherwise when trying to use the app in testing an error saying
+there are pending migrations will appear.
+
+### Target only content related to the application you are affecting
+
+Be as specific as possible when updating content, for example, include a
+condition on the `publishing_app` even if it seems unlikely there would be
+content in any other state.
+
+### Don't assume the records you are altering exist
+
+Migrations get run in test environments, on old databases or when a developer
+first clones the project. You should always make sure that your migrations run
+fine without the data there that you are expecting.
+
+### Do you need to represent this data downstream?
+
+Representing the data downstream won't work on local machines, so you will need
+to have a check in the migration making sure it is running in the right
+environment. Alternatively, and ideally, you can use the
+`represent_downstream:` class of rake tasks in Jenkins to achieve the same
+result. If you do decide to represent downstream in the migration itself, you
+will need to have `disable_ddl_transaction!` at the top to make sure the data
```

Comment:
"you must disable transactions for this migration (running the `disable_ddl_transaction!` method) as otherwise you will be representing downstream before the data is committed to the database."

Was hoping to find a link for usage of disable_ddl_transaction! but there doesn't seem to be a good one.